clar2wasm/src/words/equal.rs

```diff
@@ -113,8 +113,16 @@ impl ComplexWord for IndexOf {
         _expr: &SymbolicExpression,
         args: &[SymbolicExpression],
     ) -> Result<(), GeneratorError> {
-        // Traverse the sequence, leaving its offset and size on the stack.
         let seq = args.get_expr(0)?;
+        // workaround to fix types in the case of elements that are themself Sequences
+        if let TypeSignature::SequenceType(SequenceSubtype::ListType(ltd)) =
+            generator.get_expr_type(seq).cloned().unwrap()
+        {
+            let (elem_ty, _) = ltd.destruct();
+            generator.set_expr_type(&args[1], elem_ty)?;
```

Comment:
Could you use a `get_expr` and handle the error instead of indexing?

Reply:
fixed